packages/riipen-ui/src/components/Container.jsx

```diff
@@ -5,70 +5,99 @@ import React from "react";
 import ThemeContext from "../styles/ThemeContext";
 import withClasses from "../utils/withClasses";
 
-class Container extends React.Component {
-  static displayName = "Container";
-
-  static propTypes = {
-    /**
-     * The content inside the container.
-     */
-    children: PropTypes.any,
-
-    /**
-     * List of additional classes to apply to this component.
-     */
-    classes: PropTypes.array,
-
-    /**
-     * Determine the max-width of the container.
-     * The container width grows with the size of the screen.
-     */
-    maxWidth: PropTypes.oneOf(["sm", "md", "lg", "xl", false])
-  };
-
-  static defaultProps = {
-    classes: [],
-    maxWidth: "lg"
-  };
-
-  static contextType = ThemeContext;
-
-  render() {
-    const { classes, children, maxWidth } = this.props;
-
-    const theme = this.context;
-
-    const className = clsx(classes);
-
-    return (
-      <React.Fragment>
-        <div className={className}>{children}</div>
-        <style jsx>{`
-          div {
-            box-sizing: border-box;
-            margin: 0 auto;
-            max-width: ${maxWidth
-              ? `${theme.breakpoints[maxWidth]}px`
-              : "100%"};
-            padding: 0 ${theme.spacing(6)}px;
-            position: relative;
-          }
+import ContainerHeader from "./ContainerHeader";
+
+const Container = ({
+  border,
+  classes,
+  children,
+  color,
+  headerProps,
+  maxWidth
+}) => {
+  const theme = React.useContext(ThemeContext);
+
+  const className = clsx("root", classes);
+
+  return (
+    <React.Fragment>
+      <div className={className}>
+        {headerProps && <ContainerHeader {...headerProps} />}
```

Comment:
conditionally render header

Reply:
I wonder if this would work better if we just passed in the header. So something like 
```
<Container header={<ContainerHeader />} />
```
instead of passing all the props to the Container, then having that pass on to the Header

Reply:
For some context, i think we should do this with the input as well. We have been adding a lot of InputLabel props to control the style, where we should probably just pass the label and hint in so we dont have to keep messing with the parent component

Reply:
Yeah that sounds good!